packager/react-native-xcode.sh

```diff
@@ -69,12 +80,43 @@ set -x
 DEST=$CONFIGURATION_BUILD_DIR/$UNLOCALIZED_RESOURCES_FOLDER_PATH
 
 if [[ "$CONFIGURATION" = "Debug" && "$PLATFORM_NAME" != "iphonesimulator" ]]; then
-  PLISTBUDDY='/usr/libexec/PlistBuddy'
-  PLIST=$TARGET_BUILD_DIR/$INFOPLIST_PATH
-  IP=$(ipconfig getifaddr en0)
-  $PLISTBUDDY -c "Add NSAppTransportSecurity:NSExceptionDomains:localhost:NSTemporaryExceptionAllowsInsecureHTTPLoads bool true" $PLIST
-  $PLISTBUDDY -c "Add NSAppTransportSecurity:NSExceptionDomains:$IP.xip.io:NSTemporaryExceptionAllowsInsecureHTTPLoads bool true" $PLIST
-  echo "$IP.xip.io" > "$DEST/ip.txt"
+  IP=$(ipconfig getifaddr en1)                            # wi-fi prioritized, since the device will use wi-fi
+  if [ -z "$IP" ]; then IP=$(ipconfig getifaddr en0) ; fi # if wi-fi doesn't have an IP address, try ethernet
+
+  if [ -z "$IP" ]; then
+    echo "error: Failed to find host IP address for development server." >&2
+    echo "Please ensure that you have a valid IP address." >&2
+    exit 2
```

Comment:
This is the one that I'm referring to in the main comment. I think it's better not to stop the build if there's no detected IP address, since the bundle is still usable at this point.